source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -40,8 +58,166 @@ class OCIOInformation
     int language = OCIO_UNSUPPORT_LANGUAGE;
     // MaterialX target
     std::string target;
+
+    // Uniform resources 
+    void updateUniformResources(OCIO::GpuShaderDescRcPtr shaderDesc);
+
+    // Texture resources 
+    void updateTextureResources(OCIO::GpuShaderDescRcPtr shaderDesc);
+
+    ColorManagementResourceMapPtr resourceMap[OCIO_RESOURCEMAP_SIZE];
 };
 
+void OCIOInformation::updateUniformResources(OCIO::GpuShaderDescRcPtr shaderDesc)
+{
+    ColorManagementResourceMap& rmap = *(resourceMap[(int)ColorManagementSystem::ResourceType::UNIFORM]);
```

Comment:
From and OCIO shader, pull out and cache uniform information.
The parsing and later GL support are based on: https://github.com/AcademySoftwareFoundation/OpenColorIO/tree/master/src/libutils/oglapphelpers